newIDE/app/src/BehaviorsEditor/Editors/Physics2Editor/index.js

```diff
@@ -0,0 +1,586 @@
+                      JSON.stringify(vertices),
+                      project
+                    );
+                    this.forceUpdate();
+                  }}
+                />
+              </ImagePreview>
+            </Line>
+            <Line>
+              <ResourceSelector
+                project={this.props.project}
+                resourceSources={this.props.resourceSources}
+                onChooseResource={this.props.onChooseResource}
+                resourceExternalEditors={this.props.resourceExternalEditors}
+                resourcesLoader={this.resourcesLoader}
+                resourceKind={'image'}
+                initialResourceName={''}
+                fullWidth={false}
+                onChange={resourceName => {
+                  this.setState({ image: resourceName });
+                  this.forceUpdate();
+                }}
+              />
+            </Line>
+          </Column>
+          {shape === 'Polygon' && (
+            <PolygonEditor
+              vertices={JSON.parse(properties.get('vertices').getValue())}
+              onChangeVertexX={(newValue, index) => {
+                let vertices = JSON.parse(
+                  properties.get('vertices').getValue()
+                );
+                vertices[index].x = newValue;
+                behavior.updateProperty(
+                  'vertices',
+                  JSON.stringify(vertices),
+                  project
+                );
+                this.forceUpdate();
+              }}
+              onChangeVertexY={(newValue, index) => {
+                let vertices = JSON.parse(
+                  properties.get('vertices').getValue()
+                );
+                vertices[index].y = newValue;
+                behavior.updateProperty(
+                  'vertices',
+                  JSON.stringify(vertices),
+                  project
+                );
+                this.forceUpdate();
+              }}
+              onAdd={() => {
+                let vertices = JSON.parse(
+                  properties.get('vertices').getValue()
+                );
+                if (vertices.length >= 8) return;
+                vertices.push({ x: 0, y: 0 });
+                behavior.updateProperty(
+                  'vertices',
+                  JSON.stringify(vertices),
+                  project
+                );
+                this.forceUpdate();
+              }}
+              onRemove={index => {
+                let vertices = JSON.parse(
+                  properties.get('vertices').getValue()
+                );
+                vertices.splice(index, 1);
+                behavior.updateProperty(
+                  'vertices',
+                  JSON.stringify(vertices),
+                  project
+                );
+                this.forceUpdate();
+              }}
+            />
+          )}
+        </Line>
+        <Line>
+          <SemiControlledTextField // Debug vertices raw content

```

Comment:
I guess this is to be removed once we've fixed the issue with vertices?

Reply:
Sure, this and the vertices array check/reset.